src/github/client/app-token-cache.ts

```diff
@@ -0,0 +1,54 @@
+import { AsymmetricAlgorithm, encodeAsymmetric } from "atlassian-jwt";
+import AuthToken, { ONE_MINUTE, TEN_MINUTES } from "./auth-token";
+
+/**
+ * Holds the GitHub app's token to authenticate as a GitHub app and refreshes it as needed.
+ *
+ * An app token allows access to GitHub's /app API only. It does not allow access to a GitHub org's data.
+ *
+ * @see https://docs.github.com/en/rest/reference/apps
+ * @see https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-a-github-app
+ */
+export default class AppTokenCache {
```

Comment:
Not really a cache, more like a generator?

Reply:
It keeps the token in memory, so we won't have to call to GitHub Rest API again.

However hence we can have multiple nodes reaching out GitHub at the same time, we still could be making many calls. (We should probably use Redis to really cache it. However let's see if it become a problem)

Reply:
On, I didn't get it. It is an App token, not an Installation token

Reply:
I'll rename it to `AppTokenHolder` since both "cache" and "generator" don't feel right.